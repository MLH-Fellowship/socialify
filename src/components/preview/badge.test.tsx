import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Badge from './badge'

test('Badge renders', () => {
  const badge = shallow(<Badge color="black" name="name1" value="value1" />)

  expect(toJson(badge)).toMatchSnapshot()
  expect(badge.hasClass('badge-wrapper')).toBe(true)
  expect(badge.find('.badge-label').text()).toStrictEqual('name1')
  expect(badge.find('.badge-value').text()).toStrictEqual('value1')
  expect(badge.find('.badge-value').prop('style')).toStrictEqual({
    backgroundColor: 'black'
  })
})
