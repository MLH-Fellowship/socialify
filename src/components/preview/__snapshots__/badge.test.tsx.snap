// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders 1`] = `
<styled.div
  className="badge-wrapper"
>
  <Styled(styled.p)
    className="badge-label"
  >
    name1
  </Styled(styled.p)>
  <styled.p
    className="badge-value"
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    value1
  </styled.p>
</styled.div>
`;
